spec/factories/access_logs.rb

```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :access_log do
+    timestamp { Time.zone.now.to_i }
+    sequence(:token_id) { |n| "1234#{n}" }
+
+    after(:build) do |access_log|
+      access_log.token = create(:token) if access_log.token_id && access_log.token.nil?
+    end
```

Comment:
D'ailleurs en vrai tu peux faire plus simple:

```suggestion
    token
```

voilà

Reply:
![image](https://github.com/etalab/admin_api_entreprise/assets/9677688/5ca17b37-4044-48d6-ac93-c52a0031cf1e)

Reply:
but why 

![image](https://github.com/etalab/admin_api_entreprise/assets/9677688/92c6c05f-3fbc-474b-89c9-c704e616220d)

Reply:
En fait t'as 2 approches: soit t'utilises directement le `token_id` soit le `token`.
Je soupçonnait ta séquence `token_id` de ne pas être affecté à `access_log.token_id` parce que `1234#{n}` n'est pas un uuid.

Généralement ça se fait comme ça:

```ruby
Factory.define
  sequence(:token_id) { SecureRandom.uuid }

  factory :access_logs do 
    token_id
  end
end
```

Et après j'ai vu cette ligne, et je me suis dit "mais ..? à quoi sert le `token_id` ici ..?", j'en ai déduis que ça ne te servait pas, d'où la simplification.

Reply:
Hmmm okay,

Ce que je ne pige pas c'est pourquoi le fait de renvoyer `token` est équivalent à `token = create(:token)`

Reply:
> Hmmm okay,
> 
> Ce que je ne pige pas c'est pourquoi le fait de renvoyer `token` est équivalent à `token = create(:token)`

Alors non, `token` c'est `build(:token)` :D 

Je t'invite à lire la doc de FactoryBot https://thoughtbot.github.io/factory_bot/summary.html